### PR TITLE
Issue #4447 - separate modules for jetty and javax websockets

### LINKED ISSUE #4447: websocket module should be split into separate modules for jetty and javax APIs (Jetty-10)
**Jetty version**
10.0.x

**Description**
WebSocket module should be split into separate modules for jetty and javax APIs.

`websocket-javax.mod` and `websocket-jetty.mod`

### PR DESCRIPTION
**Pull request by lachlan-roberts**
Closes #4447

### CODE REVIEW

File: jetty-websocket/websocket-javax-server/src/main/config/modules/websocket-javax.mod [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
-# websocket client needs jetty-client
 client
-# javax.websocket needs annotations
 annotations
 
 [lib]
-lib/websocket/*.jar
+lib/websocket/websocket-core-${jetty.version}.jar
+lib/websocket/websocket-servlet-${jetty.version}.jar
+lib/websocket/jetty-javax-websocket-api-1.1.2.jar
```
Comment (lachlan-roberts) [inline jetty-websocket/websocket-javax-server/src/main/config/modules/websocket-javax.mod:16]: This seems a bit fragile, I can see this not being changed when the api version gets updated.

Comment (olamy) [reply jetty-websocket/websocket-javax-server/src/main/config/modules/websocket-javax.mod:16]: I agree. There are few modules with the same issue...
I have started fixing that with 9.4.x (branch jetty-9.4.x-config_use_placeholder_for_version)
I was a bit conservative to make such big change.
But I can do it for 10.

**Review by olamy — APPROVED** [review]
LGTM

File: jetty-websocket/websocket-jetty-server/src/main/config/modules/websocket-jetty.mod [hunk 2]
```diff
@@ -0,0 +1,21 @@
+DO NOT EDIT - See: https://www.eclipse.org/jetty/documentation/current/startup-modules.html
+
+[description]
+Enable jetty websocket for deployed web applications.
```
Comment (gregw) [inline jetty-websocket/websocket-jetty-server/src/main/config/modules/websocket-jetty.mod]: probably should be clear this is the jetty API and not just the jetty websocket implementation.

**Review by gregw — APPROVED** [review]
LGTM other than one comment change

**Pull request merged**